.circleci/scripts/config-template.yml

```diff
@@ -203,17 +203,27 @@ jobs: # Each project will have individual jobs for each specific task it has to
           name: Exit if External PR
           shell: bash.exe
           command: if [ "$CIRCLE_PR_REPONAME" ]; then circleci-agent step halt; fi
-      - unless: # Build installers unsigned on non-tagged builds
-          condition: << pipeline.git.tag >>
+      - unless: # Build installers unsigned on non-tagged builds that aren't in main or dev
+          condition: 
+            and:
+              -  << pipeline.git.tag >>
+              - or:
+                -  equal: [ main, << pipeline.git.branch >> ]
+                -  equal: [ dev, << pipeline.git.branch >> ]
```

Comment:
Maybe just signing from `main` will do, any releases from `dev` should be pre-releases anyway.